lib/src/widgets/movies_grid_view.dart

```diff
@@ -0,0 +1,99 @@
+import 'package:animate_do/animate_do.dart';
+import 'package:flutter/material.dart';
+import '../ui/movie_detail.dart';
+import '../models/movie.dart';
+import '../utils/movie_dimensions.dart';
+
+class MoviesGridView extends StatefulWidget {
+  final Stream<Movie> allMovies;
+
+  const MoviesGridView({
+    Key? key,
+    required this.allMovies,
+  }) : super(key: key);
+
+  @override
+  _MoviesGridViewState createState() => _MoviesGridViewState();
+}
+
+class _MoviesGridViewState extends State<MoviesGridView> {
+  @override
+  Widget build(BuildContext context) {
+    return StreamBuilder(
+      stream: widget.allMovies,
+      builder: (
+        context,
+        AsyncSnapshot<Movie> snapshot,
+      ) {
+        return snapshot.hasData
+            ? buildList(
+                snapshot,
+              )
+            : Center(
+                child: CircularProgressIndicator(
+                  backgroundColor: Colors.red,
+                  color: Colors.black,
+                ),
+              );
+      },
+    );
+  }
+}
+
+Widget buildList(AsyncSnapshot snapshot) {
```

Comment:
do you think this can be moved to another widget?

Reply:
buildList is now in 'movies_grid_builder' file, thank you.